lib/proxy/router.go

```diff
@@ -229,7 +229,10 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 	}
 	span.AddEvent("retrieved target server")
 
-	principals := []string{host}
+	principals := []string{
```

Comment:
I think we want to add checks to skip adding duplicates to principals.

Reply:
I have added a